Forms/MainForm.cs

```diff
@@ -656,7 +699,19 @@ private void cVC_CheckedChanged(object sender, EventArgs e)
 
         private void mExit_Click(object sender, EventArgs e)
         {
-            Application.Exit();
+            StartClose();
+        }
+
+        protected override void OnFormClosing(FormClosingEventArgs e)
+        {
+            base.OnFormClosing(e);
+            StartClose();
+            
+        }
+
+        private void StartClose()
+        {
+            WritePreset(ROOT + "settings");
```

Comment:
Why does the parameter need to full path, but not the extension? Shouldn't the WritePreset function automatically handle the path and extension?